hpcgap/lib/oper.g

```diff
@@ -1179,33 +1186,35 @@ BIND_GLOBAL( "OPER_SetupAttribute", function(getter, flags, mutflag, filter, ran
           return;
       end);
 
-
-BIND_GLOBAL( "NewAttribute", function ( arg )
-    local   name, filter, flags, mutflag, getter, rank;
-
-    # construct getter, setter and tester
-    name   := arg[1];
-    filter := arg[2];
+# construct getter, setter and tester
+BIND_GLOBAL( "NewAttribute", function ( name, filter, args... )
+    local  flags, mutflag, getter, rank;
+
+    rank := 1;
+    mutflag := false;
+    if LEN_LIST(args) = 1 and args[1] in [ "mutable", true ] then
+        mutflag := true;
+    elif LEN_LIST(args) = 1 and IS_INT(args[1]) then
+        rank := args[1];
+    elif LEN_LIST(args) = 2 and args[1] in [ "mutable", true ] and IS_INT(args[2]) then
+        mutflag := true;
+        rank := args[2];
+    elif LEN_LIST(args) > 2 then
```

Comment:
I think the logic is wrong here.. I think line 1202 should just be "else", otherwise if you get one or two arguments that don't match one of the earlier if/elifs, we silently accept them and do nothing?